sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,1033 @@
+                .getQueryFields()
+                .stream()
+                .collect(Collectors.<String, String, Float>toMap(s -> s, v -> 1.0f));
+        boolQueryBuilder.must(sqsb.fields(searchFields));
+      }
+    }
+    sourceBuilder.from(searchDTO.getOffset() != null ? searchDTO.getOffset() : 0);
+    sourceBuilder.size(searchDTO.getLimit() != null ? searchDTO.getLimit() : 250);
+    // check mode and set constraints
+    Map<String, Float> constraintsMap = ElasticSearchHelper.getConstraints(searchDTO);
+    // apply additional properties
+    if (searchDTO.getAdditionalProperties() != null
+        && searchDTO.getAdditionalProperties().size() > 0) {
+      for (Map.Entry<String, Object> entry : searchDTO.getAdditionalProperties().entrySet()) {
+        ElasticSearchHelper.addAdditionalProperties(boolQueryBuilder, entry, constraintsMap);
+      }
+    }
+    sourceBuilder.query(boolQueryBuilder);
+    SearchRequest searchRequest = new SearchRequest(indexTypeMap.get(JsonKey.INDEX));
+    searchRequest.source(sourceBuilder);
+    ActionListener<SearchResponse> listener =
+        new ActionListener<SearchResponse>() {
+          @Override
+          public void onResponse(SearchResponse searchResponse) {
+            List<Map<String, Object>> mapList = new ArrayList<>();
+            Map<String, Object> responseMap = new HashMap<>();
+            SearchHits hits = searchResponse.getHits();
+            for (SearchHit hit : hits.getHits()) {
+              mapList.add(hit.getSourceAsMap());
+            }
+            responseMap.put(JsonKey.CONTENT, mapList);
+            responseMap.put(JsonKey.COUNT, hits.getTotalHits());
+            promise.success(responseMap);
+          }
+
+          @Override
+          public void onFailure(Exception e) {
+            promise.failure(e);
+          }
+        };
+    ConnectionManager.getRestClient().searchAsync(searchRequest, listener);
+    return promise.future();
+  }
+
+  /**
+   * This method will put a new data entry inside Elastic search. identifier value becomes _id
+   * inside ES, so every time provide a unique value while saving it.
+   *
+   * @param index String ES index name
+   * @param type String ES type name
+   * @param identifier ES column identifier as an String
+   * @param data Map<String,Object>
+   * @return String identifier for created data
+   */
+  @Override
+  public Future<String> createData(
+      String index, String type, String identifier, Map<String, Object> data) {
+    long startTime = System.currentTimeMillis();
+    Promise<String> promise = Futures.promise();
+    ProjectLogger.log(
+        "ElasticSearchUtilRest createData method started at ==" + startTime + " for Type " + type,
+        LoggerEnum.PERF_LOG);
+    if (StringUtils.isBlank(identifier)
+        || StringUtils.isBlank(type)
+        || StringUtils.isBlank(index)) {
+      ProjectLogger.log("Identifier value is null or empty ,not able to save data.");
+      promise.success("ERROR");
```

Comment:
it should be promise.failure

Reply:
it only throws errors, we check for "Error" after using this method to send response.